pkg/controlplane/instance.go

```diff
@@ -258,7 +258,7 @@ func (c *Config) createLeaseReconciler() reconcilers.EndpointReconciler {
 	if err != nil {
 		klog.Fatalf("Error determining service IP ranges: %v", err)
 	}
-	leaseStorage, _, err := storagefactory.Create(*config)
+	leaseStorage, _, err := storagefactory.Create(*config, nil)
```

Comment:
Instead of changing every call site, can the new parameter be a member of the config struct?

Reply:
This is a good point - will fix tomorrow.

Reply:
Actually - this is part of `createLeaseResonciler()` method - this is basically callled once in the kube-apiserver lifetime.
So I don't think it makes sense to change anything.

Reply:
I meant to include all the "generic.NewRawStorage(config, nil)" calls, too. If you think it's not worth it, that's fine.

It still seems weird to me to have a config thingy, and an option that's not part of the config thingy. What is the rule for adding to the config thingy vs adding another parameter?

Reply:
Technically it may be possible to make it a part of config.

That said I looked deeper into the code (this time again), and piping the necessary information down to storageConfig (if we do that, we shouldn't pipe newFunc, but rather groupKind and use ObjectCreator from runtime.Scheme to create object) and that would be at least couple hundreds line of code (and e.g. path for CRDs is different than for built-in resources there).

I'm happy to look at this, but given this commit is actually fairly simple and relatively small (and almost all changes are in tests), I would prefer to not block this PR on it, so that we can get more soaking time of the feature itself.

If you're fine with this, I'm happy to open an issue and assign to myself to fix that once this one is merged.

Reply:
What you have sounds better than plumbing it everywhere. I guess I was expecting something like, this code makes a shallow copy of the config and sets the additional field, so it'd be about the same change as what you have here, just call sites that are happy with `nil` wouldn't need to change at all.

But I wouldn't insist on this or anything, it might not be an improvement.